packages/peregrine/lib/talons/Gallery/useGalleryItem.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
+
+/**
+ *
+ * @param {Object} props
+ * @param {Object} props.item - the product item being rendered
+ * @param {Object} props.queries - the queries for the gallery item
+ */
+export const useGalleryItem = props => {
+    const { item, queries } = props;
+
+    const { prefetchProductQuery } = queries;
+    const { url_key } = item;
+
+    // Prefetch the underlying product for this gallery item.
+    const [runPrefetchQuery] = useLazyQuery(prefetchProductQuery);
+    useEffect(() => {
+        if (item) {
+            runPrefetchQuery({
```

Comment:
It hurts because we potentially already have this product data elsewhere in the cache but this is more developer-friendly / less magic.

Reply:
We could use `cache-first` here which would return the cached data for the query if it exists and only fall back to network if it doesn't.

Reply:
I added cache-first in c51624861 but I'd like to talk through it in team time.